common/lib/xmodule/xmodule/modulestore/tests/test_publish.py

```diff
@@ -139,3 +157,1222 @@ def test_publish_draft_delete(self):
+        self.course = None
+
+        # For convenience, maintain a list of (block_type, block_id) pairs for all verticals/units.
+        self.all_verticals = []
+        self.all_units = []
+
+        # Course block database is keyed on (block_type, block_id) pairs.
+        # It's built during the course creation below and contains all the parent/child
+        # data needed to check the OLX.
+        self.course_db = {}
+
+        super(UniversalTestSetup, self).setUp()
+
+
+class OLXFormatChecker(unittest.TestCase):
+    """
+    Examines the on-disk course export to verify that specific items are present/missing
+    in the course export.
+    Currently assumes that the course is broken up into different subdirs.
+
+    Requires from subclasses:
+        self.root_export_dir - absolute root directory of course exports
+        self.export_dir - top-level course export directory name
+    """
+    unittest.TestCase.longMessage = True
+
+    def _get_course_export_dir(self):
+        """
+        Ensure that the course has been exported and return course export dir.
+        """
+        block_path = os.path.join(self.root_export_dir, self.export_dir)  # pylint: disable=no-member
+        self.assertTrue(
+            os.path.isdir(block_path),
+            msg='{} is not a dir.'.format(block_path)
+        )
+        return block_path
+
+    def _get_block_type_path(self, course_export_dir, block_type, draft):
+        """
+        Return the path to the block type subdirectory, factoring in drafts.
+        """
+        block_path = course_export_dir
+        if draft:
+            block_path = os.path.join(block_path, 'drafts')
+        return os.path.join(block_path, block_type)
+
+    def _get_block_filename(self, block_id):
+        """
+        Return the course export filename for a block.
+        """
+        return '{}.xml'.format(block_id)
+
+    def _get_block_contents(self, block_subdir_path, block_id):
+        """
+        Determine the filename containing the block info.
+        Return the file contents.
+        """
+        block_file = self._get_block_filename(block_id)
+        block_file_path = os.path.join(block_subdir_path, block_file)
+        self.assertTrue(
+            os.path.isfile(block_file_path),
+            msg='{} is not an existing file.'.format(block_file_path)
+        )
+        with open(block_file_path, "r") as file_handle:
+            return file_handle.read()
+
+    def _assert_parsed_xml(self, block_contents, checklist):
+        """
+        Using a dictionary with the following format:
+        {   'tag' : block_type,
+            'attrs' : attrs,
+            'children' : {
+                'tag' : child_type,
+                'attrs' : {'url_name' : child_id_regex},
+            }
+        }
+        , verify the XML of a particular block in a course export.
+        In the above dictionary, 'attrs' is a dict with {attribute:regex} pairs.
```

Comment:
It might be worth noting that setting `attrs` says to check that those attrs exist (but that there may be others), while `attrs` being falsy (`{}`/`None`) says to assert that no attributes exist.

Separately: Would it be worth splitting the various XML assertion methods up? `assertTag`, `assertAttrsContain`, `assertAttrsEmpty`, `assertAttrsEqual`, `assertChildren`?